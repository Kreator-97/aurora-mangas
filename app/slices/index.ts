export { default as uiSlice } from './uiSlice'
export { default as authSlice } from './authSlice'
export { default as shoppingCartSlice } from './shoppingCartSlice'
