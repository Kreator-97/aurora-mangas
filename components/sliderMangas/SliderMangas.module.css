.content {
  display: grid;
  grid-template-columns: 1fr;
  justify-content: center;
  max-width: var(--max);
  gap: 16px;
}

.content.single-element {
  display: flex;
  justify-content: center;
}

.content.single-element div {
  flex-grow: 1;
}

@media (min-width: 532px ) {
  .content {
    grid-template-columns: 1fr 1fr;
  }
}

@media (min-width: 720px ) {
  .content {
    display: flex;
    overflow-y: scroll;
    justify-content: flex-start;
  }
}
