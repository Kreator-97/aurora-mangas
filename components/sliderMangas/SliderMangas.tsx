import { FC, useEffect, useRef, useState } from 'react'
import { HiArrowCircleLeft, HiArrowCircleRight } from 'react-icons/hi'

import { Manga } from '../../interfaces'
import { CardManga } from '../cardManga'
import styles from './SliderMangas.module.css'

interface Props {
  mangas: Manga[]
}

const minCardWidth = 225
const gap = 16

export const SliderMangas: FC<Props> = ({mangas}) => {
  const scrollRef = useRef<HTMLDivElement>(null)
  const [ itemsPosition, setItemsPosition ] = useState<'center' | 'flex-start'>('center')
  const [ showArrows, setShowArrows ] = useState<boolean>(true)

  const onScrollLeft = () => {
    const clientWidth = scrollRef.current?.clientWidth
    const offset = (clientWidth ? clientWidth - (minCardWidth / 2) : minCardWidth) * -1
    scrollRef.current!.scrollBy( offset, 0)
  }

  const onScrollRight = () => {
    const clientWidth = scrollRef.current?.clientWidth
    scrollRef.current?.scrollBy(clientWidth ? clientWidth - (minCardWidth / 2) : minCardWidth, 0)
  }

  const computePosition = () => {
    const currentWidth = (minCardWidth * mangas.length + mangas.length * gap)
    const position = currentWidth > (scrollRef.current?.clientWidth || 0)
      ? 'flex-start'
      : 'center'
    setItemsPosition(position)
  }

  const computeShowArrows = () => {
    const currentWidth = (minCardWidth * mangas.length + mangas.length * gap)
    const showArrows = currentWidth > (scrollRef.current?.clientWidth || 0)
    setShowArrows(showArrows)
  }

  useEffect(() => {
    computePosition()
    computeShowArrows()
  }, [mangas]) // eslint-disable-line

  useEffect(() => {
    const resizeEvent = () => {
      computePosition()
      computeShowArrows()
    }

    window.addEventListener('resize', resizeEvent)
    return () => {
      window.removeEventListener('resize', resizeEvent)
    }
  }, []) // eslint-disable-line

  return (
    <section className='px-4 relative justify-center'>
      <h2 className='text-center text-xl mb-2'>{mangas.length} Mangas</h2>
      <div
        className={`${styles.content} ${mangas.length === 1 ? `${styles['single-element']}` : ''}`}
        ref={scrollRef}
        style={{
          scrollBehavior: 'smooth',
          justifyContent: itemsPosition,
        }}
      >
        {
          mangas.map((manga) => {
            return (
              <CardManga key={manga.id} manga={manga}/>
            )
          })
        }
      </div>
      <div
        className={`bg-white absolute left-0 top-1/2 rounded-full hidden md:${showArrows ? 'block' : 'hidden'}`}
      >
        <HiArrowCircleLeft
          size={28}
          onClick={() => onScrollLeft() }
          fill={'var(--accent)'}
          color="#fff"
          cursor="pointer"
          style={{ transform: 'scale(2)'}}
          stroke="var(--dark)"
          strokeWidth={.5}
          strokeOpacity={.5}
          data-testid="scroll-left-icon"
        />
      </div>
      <div
        className={`bg-white absolute right-0 top-1/2 rounded-full hidden md:${showArrows ? 'block' : 'hidden'}`}
      >
        <HiArrowCircleRight
          size={28}
          onClick={() => onScrollRight() }
          fill={'var(--accent)'}
          cursor="pointer"
          style={{ transform: 'scale(2)'}}
          className="shadow-inner"
          stroke="var(--dark)"
          strokeWidth={.5}
          strokeOpacity={.3}
          data-testid="scroll-right-icon"
        />
      </div>
    </section>
  )
}
