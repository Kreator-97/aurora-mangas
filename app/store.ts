import { configureStore } from '@reduxjs/toolkit'
import { uiSlice, authSlice, shoppingCartSlice } from './slices'

export const store = configureStore({
  reducer: {
    ui: uiSlice,
    auth: authSlice,
    cart: shoppingCartSlice
  },
})

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch
