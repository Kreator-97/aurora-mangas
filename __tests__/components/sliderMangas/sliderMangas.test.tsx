import { fireEvent, render, screen } from '@testing-library/react'
import { Provider } from 'react-redux'

import { mangas } from '../../fixtures/db'
import { SliderMangas } from '../../../components/sliderMangas/SliderMangas'
import { store } from '../../../app/store'

const scrollMock = jest.fn()
jest.mock('react', () => {
  return {
    ...jest.requireActual('react'),
    useRef: () => ({current: Object.defineProperty( HTMLDivElement.prototype, 'scrollBy', {
      configurable: true,
      value: scrollMock
    })})
  }
})

describe('test on SliderMangas component', () => {
  test('should to match with snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <SliderMangas mangas={mangas} />
      </Provider>
    )

    expect( container ).toMatchSnapshot()
    expect( screen.getByText(`${mangas.length} Mangas`)).toBeInTheDocument
    expect( screen.getAllByTestId('card-manga').length ).toBe(mangas.length)
  })

  test('should to call onScrollRight and onScrollLeft functions', () => {

    render(
      <Provider store={store}>
        <SliderMangas mangas={mangas} />
      </Provider>
    )

    const scrollRightIcon = screen.getByTestId('scroll-right-icon')
    const scrollLeftIcon = screen.getByTestId('scroll-left-icon')

    fireEvent.click(scrollRightIcon)
    expect(scrollMock).toHaveBeenCalledWith(expect.any(Number), 0)
    
    fireEvent.click(scrollLeftIcon)
    expect(scrollMock).toHaveBeenCalledWith(expect.any(Number), 0)
    expect(scrollMock).toHaveBeenCalledTimes(2)
  })
})
