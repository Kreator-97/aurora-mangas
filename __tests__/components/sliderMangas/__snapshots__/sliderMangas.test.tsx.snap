// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on SliderMangas component should to match with snapshot 1`] = `
<div>
  <section
    class="px-4 relative justify-center"
  >
    <h2
      class="text-center text-xl mb-2"
    >
      12
       Mangas
    </h2>
    <div
      class="content "
      style="scroll-behavior: smooth; justify-content: flex-start;"
    >
      <div
        class="bg-white relative min-w-[225px] max-w-[450px] rounded w-full"
        data-testid="card-manga"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 150% 0px 0px 0px;"
          />
          <img
            class="object-contain cursor-pointer rounded-t"
            data-nimg="responsive"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="p-2 flex flex-col items-center"
        >
          <a
            class="text-dark text-xl text-center hover:underline"
            href="/serie/one-punch-man"
          >
            One Punch Man
             #
            12
          </a>
          <p
            class="text-dark text-lg"
          >
            $99.00
          </p>
          <button
            class="btn bg-accent text-dark w-full"
            role="button"
          >
            Agregar al carrito
          </button>
        </div>
        <div
          class="absolute bg-dark bg-opacity-80 top-2/4 right-2 rounded-full"
        >
          <p
            class="text-2xl px-2 py-4 w-16 text-center"
          >
            # 
            12
          </p>
        </div>
        <div
          style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
        />
      </div>
      <div
        class="bg-white relative min-w-[225px] max-w-[450px] rounded w-full"
        data-testid="card-manga"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 150% 0px 0px 0px;"
          />
          <img
            class="object-contain cursor-pointer rounded-t"
            data-nimg="responsive"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="p-2 flex flex-col items-center"
        >
          <a
            class="text-dark text-xl text-center hover:underline"
            href="/serie/one-punch-man"
          >
            One Punch Man
             #
            8
          </a>
          <p
            class="text-dark text-lg"
          >
            $99.00
          </p>
          <button
            class="btn bg-accent text-dark w-full"
            role="button"
          >
            Agregar al carrito
          </button>
        </div>
        <div
          class="absolute bg-dark bg-opacity-80 top-2/4 right-2 rounded-full"
        >
          <p
            class="text-2xl px-2 py-4 w-16 text-center"
          >
            # 
            8
          </p>
        </div>
        <div
          style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
        />
      </div>
      <div
        class="bg-white relative min-w-[225px] max-w-[450px] rounded w-full"
        data-testid="card-manga"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 150% 0px 0px 0px;"
          />
          <img
            class="object-contain cursor-pointer rounded-t"
            data-nimg="responsive"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="p-2 flex flex-col items-center"
        >
          <a
            class="text-dark text-xl text-center hover:underline"
            href="/serie/one-punch-man"
          >
            One Punch Man
             #
            5
          </a>
          <p
            class="text-dark text-lg"
          >
            $99.00
          </p>
          <button
            class="btn bg-accent text-dark w-full"
            role="button"
          >
            Agregar al carrito
          </button>
        </div>
        <div
          class="absolute bg-dark bg-opacity-80 top-2/4 right-2 rounded-full"
        >
          <p
            class="text-2xl px-2 py-4 w-16 text-center"
          >
            # 
            5
          </p>
        </div>
        <div
          style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
        />
      </div>
      <div
        class="bg-white relative min-w-[225px] max-w-[450px] rounded w-full"
        data-testid="card-manga"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 150% 0px 0px 0px;"
          />
          <img
            class="object-contain cursor-pointer rounded-t"
            data-nimg="responsive"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="p-2 flex flex-col items-center"
        >
          <a
            class="text-dark text-xl text-center hover:underline"
            href="/serie/one-punch-man"
          >
            One Punch Man
             #
            2
          </a>
          <p
            class="text-dark text-lg"
          >
            $99.00
          </p>
          <button
            class="btn bg-accent text-dark w-full"
            role="button"
          >
            Agregar al carrito
          </button>
        </div>
        <div
          class="absolute bg-dark bg-opacity-80 top-2/4 right-2 rounded-full"
        >
          <p
            class="text-2xl px-2 py-4 w-16 text-center"
          >
            # 
            2
          </p>
        </div>
        <div
          style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
        />
      </div>
      <div
        class="bg-white relative min-w-[225px] max-w-[450px] rounded w-full"
        data-testid="card-manga"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 150% 0px 0px 0px;"
          />
          <img
            class="object-contain cursor-pointer rounded-t"
            data-nimg="responsive"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="p-2 flex flex-col items-center"
        >
          <a
            class="text-dark text-xl text-center hover:underline"
            href="/serie/one-punch-man"
          >
            One Punch Man
             #
            1
          </a>
          <p
            class="text-dark text-lg"
          >
            $99.00
          </p>
          <button
            class="btn bg-accent text-dark w-full"
            role="button"
          >
            Agregar al carrito
          </button>
        </div>
        <div
          class="absolute bg-dark bg-opacity-80 top-2/4 right-2 rounded-full"
        >
          <p
            class="text-2xl px-2 py-4 w-16 text-center"
          >
            # 
            1
          </p>
        </div>
        <div
          style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
        />
      </div>
      <div
        class="bg-white relative min-w-[225px] max-w-[450px] rounded w-full"
        data-testid="card-manga"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 150% 0px 0px 0px;"
          />
          <img
            class="object-contain cursor-pointer rounded-t"
            data-nimg="responsive"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="p-2 flex flex-col items-center"
        >
          <a
            class="text-dark text-xl text-center hover:underline"
            href="/serie/one-punch-man"
          >
            One Punch Man
             #
            10
          </a>
          <p
            class="text-dark text-lg"
          >
            $99.00
          </p>
          <button
            class="btn bg-accent text-dark w-full"
            role="button"
          >
            Agregar al carrito
          </button>
        </div>
        <div
          class="absolute bg-dark bg-opacity-80 top-2/4 right-2 rounded-full"
        >
          <p
            class="text-2xl px-2 py-4 w-16 text-center"
          >
            # 
            10
          </p>
        </div>
        <div
          style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
        />
      </div>
      <div
        class="bg-white relative min-w-[225px] max-w-[450px] rounded w-full"
        data-testid="card-manga"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 150% 0px 0px 0px;"
          />
          <img
            class="object-contain cursor-pointer rounded-t"
            data-nimg="responsive"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="p-2 flex flex-col items-center"
        >
          <a
            class="text-dark text-xl text-center hover:underline"
            href="/serie/one-punch-man"
          >
            One Punch Man
             #
            7
          </a>
          <p
            class="text-dark text-lg"
          >
            $99.00
          </p>
          <button
            class="btn bg-accent text-dark w-full"
            role="button"
          >
            Agregar al carrito
          </button>
        </div>
        <div
          class="absolute bg-dark bg-opacity-80 top-2/4 right-2 rounded-full"
        >
          <p
            class="text-2xl px-2 py-4 w-16 text-center"
          >
            # 
            7
          </p>
        </div>
        <div
          style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
        />
      </div>
      <div
        class="bg-white relative min-w-[225px] max-w-[450px] rounded w-full"
        data-testid="card-manga"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 150% 0px 0px 0px;"
          />
          <img
            class="object-contain cursor-pointer rounded-t"
            data-nimg="responsive"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="p-2 flex flex-col items-center"
        >
          <a
            class="text-dark text-xl text-center hover:underline"
            href="/serie/one-punch-man"
          >
            One Punch Man
             #
            4
          </a>
          <p
            class="text-dark text-lg"
          >
            $99.00
          </p>
          <button
            class="btn bg-accent text-dark w-full"
            role="button"
          >
            Agregar al carrito
          </button>
        </div>
        <div
          class="absolute bg-dark bg-opacity-80 top-2/4 right-2 rounded-full"
        >
          <p
            class="text-2xl px-2 py-4 w-16 text-center"
          >
            # 
            4
          </p>
        </div>
        <div
          style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
        />
      </div>
      <div
        class="bg-white relative min-w-[225px] max-w-[450px] rounded w-full"
        data-testid="card-manga"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 150% 0px 0px 0px;"
          />
          <img
            class="object-contain cursor-pointer rounded-t"
            data-nimg="responsive"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="p-2 flex flex-col items-center"
        >
          <a
            class="text-dark text-xl text-center hover:underline"
            href="/serie/one-punch-man"
          >
            One Punch Man
             #
            11
          </a>
          <p
            class="text-dark text-lg"
          >
            $99.00
          </p>
          <button
            class="btn bg-accent text-dark w-full"
            role="button"
          >
            Agregar al carrito
          </button>
        </div>
        <div
          class="absolute bg-dark bg-opacity-80 top-2/4 right-2 rounded-full"
        >
          <p
            class="text-2xl px-2 py-4 w-16 text-center"
          >
            # 
            11
          </p>
        </div>
        <div
          style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
        />
      </div>
      <div
        class="bg-white relative min-w-[225px] max-w-[450px] rounded w-full"
        data-testid="card-manga"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 150% 0px 0px 0px;"
          />
          <img
            class="object-contain cursor-pointer rounded-t"
            data-nimg="responsive"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="p-2 flex flex-col items-center"
        >
          <a
            class="text-dark text-xl text-center hover:underline"
            href="/serie/one-punch-man"
          >
            One Punch Man
             #
            9
          </a>
          <p
            class="text-dark text-lg"
          >
            $99.00
          </p>
          <button
            class="btn bg-accent text-dark w-full"
            role="button"
          >
            Agregar al carrito
          </button>
        </div>
        <div
          class="absolute bg-dark bg-opacity-80 top-2/4 right-2 rounded-full"
        >
          <p
            class="text-2xl px-2 py-4 w-16 text-center"
          >
            # 
            9
          </p>
        </div>
        <div
          style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
        />
      </div>
      <div
        class="bg-white relative min-w-[225px] max-w-[450px] rounded w-full"
        data-testid="card-manga"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 150% 0px 0px 0px;"
          />
          <img
            class="object-contain cursor-pointer rounded-t"
            data-nimg="responsive"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="p-2 flex flex-col items-center"
        >
          <a
            class="text-dark text-xl text-center hover:underline"
            href="/serie/one-punch-man"
          >
            One Punch Man
             #
            3
          </a>
          <p
            class="text-dark text-lg"
          >
            $99.00
          </p>
          <button
            class="btn bg-accent text-dark w-full"
            role="button"
          >
            Agregar al carrito
          </button>
        </div>
        <div
          class="absolute bg-dark bg-opacity-80 top-2/4 right-2 rounded-full"
        >
          <p
            class="text-2xl px-2 py-4 w-16 text-center"
          >
            # 
            3
          </p>
        </div>
        <div
          style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
        />
      </div>
      <div
        class="bg-white relative min-w-[225px] max-w-[450px] rounded w-full"
        data-testid="card-manga"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 150% 0px 0px 0px;"
          />
          <img
            class="object-contain cursor-pointer rounded-t"
            data-nimg="responsive"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="p-2 flex flex-col items-center"
        >
          <a
            class="text-dark text-xl text-center hover:underline"
            href="/serie/one-punch-man"
          >
            One Punch Man
             #
            6
          </a>
          <p
            class="text-dark text-lg"
          >
            $99.00
          </p>
          <button
            class="btn bg-accent text-dark w-full"
            role="button"
          >
            Agregar al carrito
          </button>
        </div>
        <div
          class="absolute bg-dark bg-opacity-80 top-2/4 right-2 rounded-full"
        >
          <p
            class="text-2xl px-2 py-4 w-16 text-center"
          >
            # 
            6
          </p>
        </div>
        <div
          style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
        />
      </div>
    </div>
    <div
      class="bg-white absolute left-0 top-1/2 rounded-full hidden md:block"
    >
      <svg
        color="#fff"
        cursor="pointer"
        data-testid="scroll-left-icon"
        fill="var(--accent)"
        height="28"
        stroke="var(--dark)"
        stroke-opacity="0.5"
        stroke-width="0.5"
        style="color: rgb(255, 255, 255); transform: scale(2);"
        viewBox="0 0 20 20"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M10 18a8 8 0 100-16 8 8 0 000 16zm.707-10.293a1 1 0 00-1.414-1.414l-3 3a1 1 0 000 1.414l3 3a1 1 0 001.414-1.414L9.414 11H13a1 1 0 100-2H9.414l1.293-1.293z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="bg-white absolute right-0 top-1/2 rounded-full hidden md:block"
    >
      <svg
        class="shadow-inner"
        cursor="pointer"
        data-testid="scroll-right-icon"
        fill="var(--accent)"
        height="28"
        stroke="var(--dark)"
        stroke-opacity="0.3"
        stroke-width="0.5"
        style="transform: scale(2);"
        viewBox="0 0 20 20"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-8.707l-3-3a1 1 0 00-1.414 1.414L10.586 9H7a1 1 0 100 2h3.586l-1.293 1.293a1 1 0 101.414 1.414l3-3a1 1 0 000-1.414z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </section>
</div>
`;
